templates/android/library/src/main/java/io/appwrite/Client.kt.twig

```diff
@@ -330,74 +332,84 @@ class Client @JvmOverloads constructor(
                 file.asRequestBody()
             )
             return call(
-                "POST",
+                method = "POST",
                 path,
                 headers,
                 params,
                 responseType,
-                convert
+                converter
             )
         }
 
-        val input = file.inputStream().buffered()
+        val input = RandomAccessFile(file, "r")
         val buffer = ByteArray(CHUNK_SIZE)
         var offset = 0L
         var result: Map<*, *>? = null
 
-        generateSequence {
-            val readBytes = input.read(buffer)
-            if (readBytes >= 0) {
-                buffer.copyOf(readBytes)
-            } else {
-                input.close()
-                null
-            }
-        }.forEach {
+        if (idParamName?.isNotEmpty() == true && params[idParamName] != "unique()") {
+            // Make a request to check if a file already exists
+            val current = call(
+                method = "GET",
+                path = "$path/${params[idParamName]}",
+                headers = headers,
+                params = emptyMap(),
+                responseType = Map::class.java,
+            )
+            val chunksUploaded = current["chunksUploaded"] as Long
+            offset = (chunksUploaded * CHUNK_SIZE).coerceAtMost(size)
+        }
+
+        while (offset < size) {
+            input.seek(offset)
+            input.read(buffer)
+
             params[paramName] = MultipartBody.Part.createFormData(
                 paramName,
                 file.name,
-                it.toRequestBody()
+                buffer.toRequestBody()
             )
 
             headers["Content-Range"] =
                 "bytes $offset-${((offset + CHUNK_SIZE) - 1).coerceAtMost(size)}/$size"
 
             result = call(
-                "POST",
+                method = "POST",
                 path,
                 headers,
                 params,
-                Map::class.java
+                responseType = Map::class.java
             )
 
             offset += CHUNK_SIZE
-            headers["x-{{ spec.title | caseLower }}-id"] = result!!["\$id"].toString()
-            onProgress?.invoke(UploadProgress(
-                id = result!!["\$id"].toString(),
-                progress = offset.coerceAtMost(size).toDouble()/size * 100,
-                sizeUploaded = offset.coerceAtMost(size),
-                chunksTotal = result!!["chunksTotal"].toString().toInt(),
-                chunksUploaded = result!!["chunksUploaded"].toString().toInt(),
-            ))
+            headers["x-appwrite-id"] = result!!["\$id"].toString()
+            onProgress?.invoke(
+                UploadProgress(
+                    id = result!!["\$id"].toString(),
```

Comment:
isn't using `!!` prone to null pointer exceptions ? Can we make it null safe ?

Reply:
It is, force-unwrapping an optional - but conveys what we want here, if there is no `result`, it is an exceptional case and we should stop uploading